src/timesync/timesyncd-manager.c

```diff
@@ -1221,3 +1223,58 @@ static int manager_save_time_and_rearm(Manager *m, usec_t t) {
 
         return 0;
 }
+
+static const char* ntp_server_property_name[] = {
+        "SystemNTPServers",
+        "FallbackNTPServers",
+        "LinkNTPServers",
+        "RuntimeNTPServers",
+};
```

Comment:
this is fragile. to make it less so:

```c
static const char* ntp_server_property_name[_SERVER_TYPE_MAX] = {
        [SERVER_SYSTEM]   = "SystemNTPServers",
        [SERVER_FALLBACK] = "FallbackNTPServers",
        …
}
```

which means the array will definitely be properly sized and entries will definitely match.

Even better: use DEFINE_STRING_TABLE_LOOKUP_TO_STRING() to define an automatic, safe accessor function to such an array
}

Reply:
thanks for your reply, modified per your good suggestion